File: 101_lessons/lesson_5/practice_problems/5.rb
# Given this nested Hash:

munsters = {
  "Herman" => { "age" => 32, "gender" => "male" },
  "Lily" => { "age" => 30, "gender" => "female" },
  "Grandpa" => { "age" => 402, "gender" => "male" },
  "Eddie" => { "age" => 10, "gender" => "male" },
  "Marilyn" => { "age" => 23, "gender" => "female"}
}

# figure out the total age of just the male members of the family.

# Nested hash - hashes inside a hash
# Select the only keys that have a value of gender="male"

men = munsters.select do |k, v|
  v["gender"] == "male"
end

# Collect the ages- can we use reduce?
ages = men.values.map { |i| i["age"] }
puts ages.reduce { |sum, age| sum + age }

File: lessons/210_lessons/lesson_2/5.js
function average(arr) {
  return sum(arr) / arr.length;
}

function sum(arr) {
  var total = 0;

  for (var i = 0; i < arr.length; i++) {
    total += arr[i];
  }

  return total;
}

var temperatures = [45, 50, 60, 54, 32];
console.log(average(temperatures));
